pkg/querier/tripperware/instantquery/instant_query_test.go

```diff
@@ -220,138 +371,785 @@ func TestMergeResponse(t *testing.T) {
 	for _, tc := range []struct {
 		name         string
 		req          tripperware.Request
-		resps        []string
+		resps        []PrometheusInstantQueryResponse
 		expectedResp string
 		expectedErr  error
 	}{
 		{
-			name:         "empty response",
-			req:          defaultReq,
-			resps:        []string{`{"status":"success","data":{"resultType":"vector","result":[]}}`},
+			name: "empty response",
+			req:  defaultReq,
+			resps: []PrometheusInstantQueryResponse{
+				{
+					Status: "success",
+					Data: PrometheusInstantQueryData{
+						ResultType: model.ValVector.String(),
+						Result: PrometheusInstantQueryResult{
+							Result: &PrometheusInstantQueryResult_Vector{
+								Vector: &Vector{
+									Samples: make([]*Sample, 0),
+								},
+							},
+						},
+					},
+					Headers: []*tripperware.PrometheusResponseHeader{
+						{Name: "Content-Type", Values: []string{"application/x-protobuf"}},
+					},
+				},
+			},
 			expectedResp: `{"status":"success","data":{"resultType":"vector","result":[]}}`,
 		},
 		{
-			name:         "empty response with stats",
-			req:          defaultReq,
-			resps:        []string{`{"status":"success","data":{"resultType":"vector","result":[],"stats":{"samples":{"totalQueryableSamples":0,"totalQueryableSamplesPerStep":[]}}}}`},
+			name: "empty response with stats",
+			req:  defaultReq,
+			resps: []PrometheusInstantQueryResponse{
+				{
+					Status: "success",
+					Data: PrometheusInstantQueryData{
+						ResultType: model.ValVector.String(),
+						Result: PrometheusInstantQueryResult{
+							Result: &PrometheusInstantQueryResult_Vector{
+								Vector: &Vector{
+									Samples: []*Sample{},
+								},
+							},
+						},
+						Stats: &tripperware.PrometheusResponseStats{
+							Samples: &tripperware.PrometheusResponseSamplesStats{
+								TotalQueryableSamplesPerStep: []*tripperware.PrometheusResponseQueryableSamplesStatsPerStep{},
+								TotalQueryableSamples:        0,
+							},
+						},
+					},
+					Headers: []*tripperware.PrometheusResponseHeader{
+						{Name: "Content-Type", Values: []string{"application/x-protobuf"}},
+					},
+				},
+			},
 			expectedResp: `{"status":"success","data":{"resultType":"vector","result":[],"stats":{"samples":{"totalQueryableSamples":0,"totalQueryableSamplesPerStep":[]}}}}`,
 		},
```

Comment:
@yeya24 these two test cases fail because the empty array is converted to nil during protobuf marshalling & unmarshalling